Group dependency updates

colincasey opened the pull request:
Adding configuration to dependabot to take advantage of [group updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups).

[GUS-W-13579117](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-13579117)

colincasey committed:
Group dependency updates

Adding configuration to dependabot to take advantage of [group updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups).

[GUS-W-13579117](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-13579117)

edmorley commented on .github/dependabot.yml:
```diff
@@ -4,3 +4,8 @@ updates:
     schedule:
       interval: "monthly"
+    groups:
+      dependencies:
```
This repo doesn't have updates enabled for GitHub Actions - I think they should be enabled too. At which point, I think it might be worth naming the group something like `java-dependencies` to differentiate them?

(IMO we don't need to enable grouping for GitHub Actions - we haven't for other repos, since on the most part the actions use the major version tags anyway, so there would be no minor/patch versions to group)

colincasey replied:
added GH action updates and renamed this group to `java-dependencies`

colincasey committed:
Group dependency updates

Adding configuration to dependabot to take advantage of [group updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups).

[GUS-W-13579117](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-13579117)

colincasey committed:
Group dependency updates

Adding configuration to dependabot to take advantage of [group updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups).

[GUS-W-13579117](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-13579117)
